src/DelegateDecompiler/ReplaceExpressionVisitor.cs

```diff
@@ -24,5 +26,18 @@ public override Expression Visit(Expression node)
             }
             return base.Visit(node);
         }
+
+        protected override Expression VisitMethodCall(MethodCallExpression node)
+        {
+            var objectType = node.Object?.Type;
+            if (objectType!=null && objectType.IsGenericType && typeof(Nullable<>) == objectType.GetGenericTypeDefinition() && node.Method.Name == "GetValueOrDefault")
+            {
+                var elementType = objectType.GetGenericArguments().First();
+                if (elementType.IsPrimitive) { 

```

Comment:
Not really sure the condition is necessary but it agrees with existing raw unit test Test05DateTime